src/modules/account/components/AddAccountForm/AddAccountForm.js

```diff
@@ -22,18 +23,42 @@ const AddAccountForm = ({ settings, onAddAccount }) => {
     });
   };
 
+  const props = { settings, onAddAccount, setPassphrase, passphrase };
+
+  if (settings.enableCustomDerivationPath) {
+    return <AddAccountFormWithDerivationPath {...props} />;
+  }
+  return <AddAccountFormContainer {...props} />;
+};
+
+const AddAccountFormContainer = ({
+  settings,
+  passphrase,
+  setPassphrase,
+  onAddAccount,
+  isSubmitDisabled,
+  derivationPath,
+  children,
+}) => {
+  const { t } = useTranslation();
+
   const onFormSubmit = (e) => {
     e.preventDefault();
     // istanbul ignore else
     if (passphrase.value && passphrase.isValid) {
-      onAddAccount(passphrase);
+      onAddAccount(passphrase, derivationPath);
```

Comment:
Now that you separated the logic, I am not sure why do we need to pass this property? an account can be created with or without derivationPath.
- With derivationPath: either default or specific
- Without derivationPath: nothing to pass